### ISSUE TITLE
Argument Parser version check

### ISSUE THREAD

**Issue by h-2** [seqan/seqan3#956]
It would likely be good if someone knowledgeable of the law analyses this, but I think if we do the following we should be *rather* safe:

1. change the server side to not store the IP-addresses, but instead immediately resolve whatever we want to store (GeoIP look-up). This really needs to change!
2. explicitly ask for confirmation upon starting the application (before invoking the thread). This could look like this:
```
#######################################################################
   Automatic Update Notifications
#######################################################################

 Applications built with SeqAn can check for updates automatically. 
 This ensures you will be informed about new versions and important
 fixes and is highly recommended. It involves making a web-request
 to a server and transmitting basic application metadata (of course
 not your data files!). For privacy implications, please see:
 https://github.com/seqan/seqan3/wiki/Update-Notifications

    [a] Always perform version checks for this app (the default).
    [n] Never perform version checks for this app.
    [o] Perform a version check now, but ask again next time.
    [s] Skip the version check now, but ask again next time.

 You can change the behaviour at any time by passing  
      --version-check X
 to your application where "X" is one of the above letters.

#######################################################################
```

We can detect whether the program is running in interactive mode or not. In the latter case we need to decide on a default behaviour. The safe option would be to not do it in non-interactive situations, but maybe we can also do it if we display a notice like before.

If we want this to be accepted by thrid parties, we should also add the possibility to easily change the update server URL. And I think we should have infrastructure server-side that let's other people easily add there applications to our database, with automatic tracking of new releases (e.g. via github-API).

**Comment by smehringer** [comment]
> change the server side to not store the IP-addresses, but instead immediately resolve whatever we want to store (GeoIP look-up). This really needs to change!

I agree but this involves setting up our own server to have more control over it (right now the version calls go to Tuebingen) and I am not gonna do it. So who will? And do we need that before the release?

**Comment by smehringer** [comment]
Resolved by #955
